code/game/turfs/simulated/floor/lava_pool.dm

```diff
@@ -0,0 +1,45 @@
+/turf/simulated/floor/lava_pool
+		icon_state = "lava"
+		name = "lava pool"
+		desc = "A deadly pool of molten rock that incinerates almost anything on contact."
+		luminosity = 3
+
+/turf/simulated/floor/lava_pool/Entered(var/mob/living/carbon/C)
+	if(!istype(C))
+		return
+
+	SSobj.processing |= src
+	if(istype(C, /mob/living/carbon/human))
+		if(check_suit(C))
+			C.visible_message("<span class='warning'>Your reinforced suit protects you from [src]! </span>")
+			return
+
+	apply_lava_damage(C)
+	C.emote("scream")
+	C.visible_message("<span class='warning'>[C] falls into[src]!   </span>", "<span class='userdanger'>You fall into [src]!   </span>", "<span class='warning'>You hear a splash followed   by a scream! </span>")
+
+/turf/simulated/floor/lava_pool/process()
+	var/mobpresent = 0
+
+	for(var/mob/living/carbon/human/H in src)
+		mobpresent = 1
```

Comment:
not if they're the only mob in the turf, it won't  readd them to a processing list unless someone else enters the turf.